CMakeLists.txt

```diff
@@ -0,0 +1,237 @@
+cmake_minimum_required(VERSION 3.5)
+
+project(OpenTTD)
+
+if (${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
+    message(FATAL_ERROR "In-source builds not allowed. Please run \"cmake ..\" from the bin directory")
+endif (${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake")
+
+include(Options REQUIRED)
+set_options()
+set_directory_options()
+
+# Make sure we have Threads available. We don't use autodetect_library here,
+# as the CMake 'FindThreads' is a lot more clever, always available, and
+# gives us a Threads::Threads dependency to link to our application.
+find_package(Threads REQUIRED)
+
+include(AutodetectLibrary REQUIRED)
+autodetect_library("zlib" ZLIB ZLIB zlib "" zlib.h z)
```

Comment:
Yes and no. It does for some libraries, it does not for others. To avoid making some library detection more special than others, we decided to just all make them equal. In reality, it will never fall back on pkg-config or manual detection for these anyway, so there is no real harm.

Mostly we noticed this with libraries that are supported in newer CMake, but not in older. There it became really blurry real quick. Given the wide range of systems we need to support, to me this feels like the most safe solution.

I am open for counter-arguments ;)

Reply:
I would go this route, too. This is easier to follow for everyone used to CMake, and you still can call your custom detect function in your own Find*.cmake if that really makes things easier.

Reply:
I'd like to see this addressed before merge. There's no reason to go all NIH for library detection, of all things.